xmrstak/backend/nvidia/nvcc_code/cuda_extra.cu

```diff
@@ -496,13 +504,14 @@ extern "C" int cuda_get_deviceinfo(nvid_ctx* ctx)
 	while ( ss >> tmpArch )
 		arch.push_back( tmpArch );
 
+	#define MSG_CUDA_NO_ARCH "WARNING: skip device - binary does not contain required device architecture\n"
```

Comment:
Please do not use a define for this. Please keep the message duplicated or use a intermediate variable.